src/server/img_storage.rs

```diff
@@ -228,6 +250,9 @@ impl StoreImage {
             .await
             .map_err(|e| HandlerErrorKind::Internal(format!("Image body error: {:?}", e)))?;
 
+        let meta = self.meta(&image)?;
```

Comment:
For context, per newtab specs, images for TopSites should be square, if `heigh != width`, we can treat it as an invalid image. Also, the minimal hi-res icon size for TopSites is 96 pixels, so if it's below that, we can also treat it as invalid.

Assuming we're going to use `image_url` for the URL, shall we call it `image_size` for this one?

Reply:
Works for me! Thanks!

are there any other things we should check the image for? (e.g. max size?, unusual format? (not "jpg" or "png") etc.)

Reply:
I think we can enforce the following things here:
* Size: `width == height && 96 <= wide <= 256`
* MIME extensions: Firefox only takes any of `png, jpg, svg`